wire-ios/Wire-iOS UnitTests/DeviceInfoViewModelTests.swift

```diff
@@ -0,0 +1,130 @@
+
+        mockConversationUserClientDetailsActions.howToDoThat_MockMethod = {
+            expectation.fulfill()
+        }
+        deviceInfoViewModel.onHowToDoThatTapped()
+        wait(for: [expectation])
+    }
+
+    func testThatItCallsDownloadCertificateMethodInDeviceActionsHandler_WhenDownloadCertificateIsCalled() {
+        let expectation = expectation(description: "copy value should be called")
+
+        mockDeviceActionsHandler.downloadE2EIdentityCertificateCertificate_MockMethod = {[weak self] value in
+            XCTAssertEqual(self?.deviceInfoViewModel.e2eIdentityCertificate, value)
+            expectation.fulfill()
+        }
+        deviceInfoViewModel.downloadE2EIdentityCertificate()
+        wait(for: [expectation])
+    }
+
+    func testThatItCallsCopyValueMethodInDeviceActionsHandler_WhenCopyValueMethodIsCalled() {
+        let copyValue = String.randomRemoteIdentifier()
+        let expectation = expectation(description: "copy value should be called")
+
+        mockDeviceActionsHandler.copyToClipboard_MockMethod = { value in
+            XCTAssertEqual(copyValue, value)
+            expectation.fulfill()
+        }
+        deviceInfoViewModel.copyToClipboard(copyValue)
+        wait(for: [expectation])
+    }
+
+    func testThatItCallsUpdateVerifiedMethodInDeviceActionsHandler_WhenUpdateVerifiedMethodIsCalled() async {
+        let isVerfied = false
+        let expectation = expectation(description: "update verified should be called")
+        mockDeviceActionsHandler.updateVerified_MockMethod = { value in
+            XCTAssertEqual(isVerfied, value)
+            expectation.fulfill()
+            return true
+        }
+        await deviceInfoViewModel.updateVerifiedStatus(isVerfied)
+        await fulfillment(of: [expectation])
+    }
+
+    func testThatItCallsRemoveDeviceMethodInDeviceActionsHandler_WhenRemoveDeviceMethodIsCalled() async {
+        let expectation = expectation(description: "removeDevice should be called")
+        mockDeviceActionsHandler.removeDevice_MockMethod = {
+            expectation.fulfill()
+            return true
+        }
+        await deviceInfoViewModel.removeDevice()
+        await fulfillment(of: [expectation])
+    }
+
+    func testThatItCallsEnrollMethodInDeviceActionsHandler_WhenEnrolClientMethodIsCalled() async {
+        let expectation = expectation(description: "enrollClient should be called")
+        mockDeviceActionsHandler.enrollClient_MockMethod = {
+            expectation.fulfill()
+            return ""
+        }
+        await deviceInfoViewModel.enrollClient()
+        await fulfillment(of: [expectation])
+    }
+
+    func testThatItCallsshowCertificateUpdateSuccess_WhenEnrolCertificateIsSuccessful() async {
+        let expectation = expectation(description: "showCertificateUpdateSuccess should be called")
+        mockDeviceActionsHandler.enrollClient_MockMethod = {
+            return ""
+        }
```

Comment:
Equal to `mockDeviceActionsHandler.enrollClient_MockValue = ""`

Reply:
done